tfhe/src/high_level_api/integers/unsigned/compressed.rs

```diff
@@ -102,15 +111,63 @@ impl<Id: FheUintId> Named for CompressedFheUint<Id> {
     const NAME: &'static str = "high_level_api::CompressedFheUint";
 }
 
+#[derive(Clone, serde::Serialize, serde::Deserialize)]
+pub enum CompressedRadixCiphertext {
+    Seeded(IntegerCompressedRadixCiphertext),
+    ModulusSwitched(CompressedModulusSwitchedRadixCiphertext),
+}
+
+impl ParameterSetConformant for CompressedRadixCiphertext {
+    type ParameterSet = RadixCiphertextConformanceParams;
+    fn is_conformant(&self, params: &RadixCiphertextConformanceParams) -> bool {
+        match self {
+            Self::Seeded(ct) => ct.is_conformant(params),
+            Self::ModulusSwitched(ct) => ct.is_conformant(params),
+        }
+    }
+}
+
+impl<Id> FheUint<Id>
+where
+    Id: FheUintId,
+{
+    pub fn compress(self) -> CompressedFheUint<Id> {
+        CompressedFheUint::new(CompressedRadixCiphertext::ModulusSwitched(
+            with_cpu_internal_keys(|sk| {
+                sk.key
+                    .switch_modulus_and_compress_parallelized(&self.ciphertext.into_cpu())
+            }),
+        ))
+    }
+}
+
 #[cfg(test)]
 mod test {
     use super::*;
     use crate::core_crypto::prelude::UnsignedInteger;
     use crate::shortint::parameters::PARAM_MESSAGE_2_CARRY_2_KS_PBS;
-    use crate::shortint::CiphertextModulus;
+    use crate::shortint::{CiphertextModulus, CompressedCiphertext};
     use crate::{generate_keys, set_server_key, CompressedFheUint8, ConfigBuilder};
     use rand::{thread_rng, Rng};
 
+    impl<Id> CompressedFheUint<Id>
+    where
+        Id: FheUintId,
+    {
+        fn blocks(&self) -> &Vec<CompressedCiphertext> {
```

Comment:
if it's needed only for the tests, I believe you should be able to match to on the enum in the test directly to avoid having functions that may panic if misused

or mark this #[cfg(test)] with still a nice error message

Reply:
ah thx